constraints.pro

```diff
@@ -267,13 +267,17 @@ gen_enforced_field(WorkspaceCwd, 'scripts.changelog:validate', ProperChangelogVa
   atomic_list_concat(['../../scripts/validate-changelog.sh ', WorkspacePackageName], ProperChangelogValidationScript).
 
 % All non-root packages must have the same "test" script.
-gen_enforced_field(WorkspaceCwd, 'scripts.test', 'jest') :-
+gen_enforced_field(WorkspaceCwd, 'scripts.test', 'jest --reporters=jest-silent-reporter') :-
   WorkspaceCwd \= '.'.
 
 % All non-root packages must have the same "test:clean" script.
 gen_enforced_field(WorkspaceCwd, 'scripts.test:clean', 'jest --clearCache') :-
```

Comment:
I noticed you changed the behavior of this script at the root level. I just realized that this script is inconsistent with how `build:clean` works. `build:clean` runs a clean build, i.e., it removes `dist` and then builds all packages. It feels like `test:clean` out to do the same thing. So maybe we should have:

``` suggestion
gen_enforced_field(WorkspaceCwd, 'scripts.test:clean', 'jest --clearCache && yarn test') :-

Reply:
The individual packages don't seem to have `build:clean` scripts. The root-level `test:clean` removes the jest cache and then runs the tests as well. Adding `&& yarn test` to the package-level `test:clean` scripts breaks the root-level `test:clean` due to the concurrent writes to the jest cache directory (see https://github.com/MetaMask/core/pull/1949#discussion_r1378120375).